### ISSUE TITLE
Heading elements are not in a sequentially-descending order

### ISSUE THREAD

**Issue by sigurdm** [dart-lang/pub-dev#7229]
It is a bit of a mess.

![image](https://github.com/dart-lang/pub-dev/assets/8613953/20f464be-e134-47ce-9756-d2b0d9e99b68)

From lighthouse:
<img width="501" alt="image" src="https://github.com/dart-lang/pub-dev/assets/8613953/8ccb82bb-d720-4611-9e0f-33b5ce055c42">

Guidelines: https://dequeuniversity.com/rules/axe/4.8/heading-order

**Comment by sigurdm** [comment]
This is also a problem for the search result page https://pub.dev/packages?q=retry

All headings are level 3. There should probably be 
* headings for the "search option" and "search results"
* a top-level heading
* Search results should have level 2
